src/ImageManipulation.php

```diff
@@ -1057,4 +1062,98 @@ protected function castDimension($value, $dimension)
         }
         return $value;
     }
+
+    private function copyImageBackend()
+    {
+        // Store result in new DBFile instance
+        /** @var DBFile $file */
+        $file = DBField::create_field(
+            'DBFile',
+            [
+                'Filename' => $this->getFilename(),
+                'Hash' => $this->getHash(),
+                'Variant' => $this->getVariant()
+            ]
+        );
+
+        $file->setImageBackend($this->getImageBackend());
+        return $file->setOriginal($this);
+    }
+
+    public function setAttribute($name, $value)
+    {
+        $file = $this->copyImageBackend();
+        $file->setMutableAttributes(array_merge(
+            $this->attributes,
+            [$name => $value]
+        ));
+
+        return $file;
+    }
+
+    public function setMutableAttributes($attributes)
+    {
+        $this->attributes = $attributes;
+    }
+
+    protected function getDefaultAttributes(): array
+    {
+        if (!$this->getIsImage()) {
+            return [];
+        }
+
+        $attributes = [
+            'width' => $this->getWidth(),
```

Comment:
Maybe those should only be added to the HTML tag if lazy loading is enabled. Not sure.

Reply:
I'd say we should just add them regardless

This might cause some images on existing sites that have their dimensions specified via CSS (or maybe not, I'm not sure on whether attributes or CSS (or width and/or height specified in style attributes) has priority

However the vast majority of images are now going to now be lazy loaded by default, which means they're getting width + height attributes added

So excluding width + height on eager loaded only images doesn't really make much sense in this context